### PR TITLE
add model endpoints to NLU, tests, new version date

### PR DESCRIPTION
**Pull request by g-may**

### CODE REVIEW

**Comment by g-may** [comment]
API reference:
[List models](https://www.ibm.com/watson/developercloud/natural-language-understanding/api/v1/#list-models)
[Delete model](https://www.ibm.com/watson/developercloud/natural-language-understanding/api/v1/#delete-model)

File: README.md [hunk 1]
```diff
@@ -411,7 +411,7 @@ var NaturalLanguageUnderstandingV1 = require('watson-developer-cloud/natural-lan
 var nlu = new NaturalLanguageUnderstandingV1({
   username: '<username>',
   password: '<password>',
-  version_date: NaturalLanguageUnderstandingV1.VERSION_DATE_2016_01_23
+  version_date: '2017-02-27'
```
Comment (nfriedly) [inline README.md]: Please use the appropriate constant here.

File: natural-language-understanding/v1.js [hunk 2]
```diff
@@ -28,15 +28,15 @@ const BaseService = require('../lib/base_service');
 function NaturalLanguageUnderstandingV1(options) {
   BaseService.call(this, options);
   if (typeof this._options.version_date === 'undefined') {
-    throw new Error('Argument error: version_date was not specified, use NaturalLanguageUnderstandingV1.VERSION_DATE_2016_01_23');
+    throw new Error('Argument error: version_date was not specified, use "2017-02-27"');
```
Comment (nfriedly) [inline natural-language-understanding/v1.js]: ditto above

File: natural-language-understanding/v1.js [hunk 3]
```diff
@@ -28,15 +28,15 @@ const BaseService = require('../lib/base_service');
...
   }
   this._options.qs.version = this._options.version_date;
 }
 util.inherits(NaturalLanguageUnderstandingV1, BaseService);
 NaturalLanguageUnderstandingV1.prototype.name = 'natural_language_understanding';
 NaturalLanguageUnderstandingV1.prototype.version = 'v1';
 NaturalLanguageUnderstandingV1.URL = 'https://gateway.watsonplatform.net/natural-language-understanding/api';
-NaturalLanguageUnderstandingV1.VERSION_DATE_2016_01_23 = '2016-01-23';
+NaturalLanguageUnderstandingV1.VERSION_DATE_2017_02_27 = '2017-02-27';
```
Comment (nfriedly) [inline natural-language-understanding/v1.js]: We can't delete the old constant, that will break users that are currently depending on it.

Also, ideally the new one should have a comment with a summary of the changes or at least a link to the relevant docs.

File: natural-language-understanding/v1.js [hunk 4]
```diff
@@ -80,4 +84,30 @@ NaturalLanguageUnderstandingV1.prototype.analyze = function(params, callback) {
   return requestFactory(parameters, callback);
 };
 
+NaturalLanguageUnderstandingV1.prototype.listModels = function(params, callback) {
```
Comment (nfriedly) [inline natural-language-understanding/v1.js]: JSDoc?

File: natural-language-understanding/v1.js [hunk 5]
```diff
@@ -80,4 +84,30 @@ NaturalLanguageUnderstandingV1.prototype.analyze = function(params, callback) {
...
+      url: '/v1/models',
+      method: 'GET',
+      json: true
+    },
+    defaultOptions: this._options
+  };
+  return requestFactory(parameters, callback);
+};
+
+NaturalLanguageUnderstandingV1.prototype.deleteModel = function(params, callback) {
```
Comment (nfriedly) [inline natural-language-understanding/v1.js]: JSDdoc

File: test/unit/test.natural_language_understanding.v1.js [hunk 6]
```diff
@@ -38,10 +38,10 @@ describe('natural_language_understanding', function() {
     done();
   });
 
-  it('analyze()', function(done) {
+  it('2017_02_27 version should work', function(done) {
```
Comment (nfriedly) [inline test/unit/test.natural_language_understanding.v1.js]: this should stay `analyze()`, although one more test that uses the old version_date would be nice.

File: test/unit/test.natural_language_understanding.v1.js [hunk 7]
```diff
@@ -51,4 +51,13 @@ describe('natural_language_understanding', function() {
 
     nlu.analyze(options, done);
   });
+
+  it('should list models', function(done) {
+    nock(watson.NaturalLanguageUnderstandingV1.URL)
+      .persist()
```
Comment (nfriedly) [inline test/unit/test.natural_language_understanding.v1.js]: I know the other one is wrong also, but `.persist()`shouldn't be called on nock's that are set up for a single test. (Actually, can you remove the other `.persist()` while you're at it?)

File: test/unit/test.natural_language_understanding.v1.js [hunk 8]
```diff
@@ -37,11 +37,23 @@ describe('natural_language_understanding', function() {
...
+    nock(watson.NaturalLanguageUnderstandingV1.URL)
+      .post('/v1/analyze?version=' + watson.NaturalLanguageUnderstandingV1.VERSION_DATE_2016_01_23)
+      .reply(200, {});
+
+    const options = {
+      features: { concepts: {}, keywords: {} },
+      text: 'hello, this is a test'
+    };
+
+    nlu.analyze(options, done);
```
Comment (nfriedly) [inline test/unit/test.natural_language_understanding.v1.js]: This nlu instance is the one with the new version date, so it won't hit the nock you set up above. You'll need to create a new instance just for this test.

If this test already passes as-is then I think we must be doing something else wrong...

**Comment by g-may** [comment]
@nfriedly changes delivered 📦

**Comment by nfriedly** [comment]
Released in v2.25.0, thanks!

**Pull request merged**